File: Dataset/Leetcode/valid/6/74.java
 class Solution {
    public String XXX(String s, int numRows) {
        if(numRows==1)
            return s;
        int i=0,poor=2*numRows-2,len=s.length();//观察示例给出的Z形可以发现每个循环中间行都比收尾行多出一个
        StringBuffer sb=new StringBuffer();
        while(i<len){//首行
            sb.append(s.charAt(i));
            i+=poor;
        }
        for(int j=1;j<numRows-1;j++){//中间行
            i=0;
            while(i<len){
                int n=i+j;
                if(n<len)
                    sb.append(s.charAt(n));
                i+=poor;
                n=i-j;
                if(n<len)
                    sb.append(s.charAt(n));
            }

        }
        i=numRows-1;
        while(i<len){//尾行
            sb.append(s.charAt(i));
            i+=poor;
        }
        return sb.toString();
    }
}

File: Dataset/Leetcode/valid/88/74.cpp
 class Solution {
public:
    void XXX(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    int i = 0, j = 0, num[m+n];
    while (i < m || j < n) {
        if(m==0){
            for(j=0;j<n;j++) num[j]=nums2[j];
            break;
        } 
        if(n==0){
            for(i=0;i<m;i++) num[i]=nums1[i];
            break;
        }
        while (i < m&&(nums1[i] <= nums2[j] || j == n)){
            num[i + j] = nums1[i];
            i++;
        } 
        while (j < n&&(nums2[j] <= nums1[i] || i == m)){
            num[i + j] = nums2[j];
            j++;
        }
    }
    for (i = 0; i < m + n; i++) nums1[i] = num[i];
    }
};

